File: evals/fixtures/flutter-security-vuln-app/lib/storage.dart
import 'dart:io';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:sqflite/sqflite.dart';
import 'package:path_provider/path_provider.dart';

// Persona: end-user auth tokens. Device thief, backup extractor, or any
// other app on a rooted device reads these — SharedPreferences is a
// plaintext XML file inside the app data dir, not Keychain/Keystore.
class SessionStore {
  Future<void> persistTokens(String accessToken, String refreshToken) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString('access_token', accessToken);
    await prefs.setString('refresh_token', refreshToken);

    final docs = await getApplicationDocumentsDirectory();
    final tokenFile = File('${docs.path}/tokens.txt');
    await tokenFile.writeAsString('access=$accessToken\nrefresh=$refreshToken');

    final db = await openDatabase('shop.db', version: 1,
        onCreate: (db, _) async {
      await db.execute(
          'CREATE TABLE session (id INTEGER PRIMARY KEY, token TEXT)');
    });
    await db.insert('session', {'token': accessToken});
  }

  Future<String?> readAccessToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString('access_token');
  }
}

File: evals/fixtures/android-code-security-vuln-app/app/src/main/java/com/example/shop/Storage.kt
package com.example.shop

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import java.io.File

// Persona: end-user auth tokens. Device thief / rooted device / adb backup
// reads all three stores — none is Keychain/Keystore backed.
class Storage(context: Context) : SQLiteOpenHelper(context, "shop.db", null, 1) {
    private val prefs = context.getSharedPreferences("session", Context.MODE_PRIVATE)

    fun persistTokens(accessToken: String, refreshToken: String) {
        prefs.edit().putString("access_token", accessToken).apply()
        prefs.edit().putString("refresh_token", refreshToken).apply()

        val tokenFile = File(context.filesDir, "tokens.txt")
        tokenFile.writeText("access=$accessToken\nrefresh=$refreshToken")
    }

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE session (id INTEGER PRIMARY KEY, token TEXT)")
    }

    fun insertToken(db: SQLiteDatabase, accessToken: String) {
        db.execSQL("INSERT INTO session (token) VALUES ('$accessToken')")
    }

    // search() input arrives from the deeplink query in MainActivity — local
    // SQLi: the value is interpolated into the statement, not bound.
    fun findCartByName(db: SQLiteDatabase, name: String) {
        db.execSQL("SELECT * FROM carts WHERE name = '$name'")
    }
}
